File: sources/api/Models/UserInGroup/UserInGroup.cs
namespace DotNetAPI.Models.UserInGroup
{
    public class UserInGroup
    {
        public int UserId { get; set; }
        public required User.User User { get; set; }
        public int GroupId { get; set; }
        public required Group.Group Group { get; set; }
        public required bool IsActive { get; set; }
        public required bool IsGroupAdmin { get; set; }
    }
}

File: sources/mobile/app/src/main/java/com/console/ratcord/domain/entity/userInGroup/UserInGroup.kt
package com.console.ratcord.domain.entity.userInGroup

import androidx.room.Entity
import androidx.room.ForeignKey
import com.console.ratcord.domain.entity.group.Group
import com.console.ratcord.domain.entity.user.User
import kotlinx.serialization.Serializable

@Serializable
@Entity(
    tableName = "UserInGroup",
    foreignKeys = [
        ForeignKey(
            entity = User::class,
            parentColumns = ["userId"],
            childColumns = ["userId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Group::class,
            parentColumns = ["groupId"],
            childColumns = ["groupId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class UserInGroup(
    val userId: Int,
    val groupId: Int,
    val isGroupAdmin: Boolean,
    val isActive: Boolean
)
